docs/proposals/hosted_components_templates.md

```diff
@@ -0,0 +1,559 @@
+# Use [OCP templates and] ABPs to install hosted components
+
+## Description
+Currently most of our hosted components have their own repositories where they
+maintain changes and then need to synchronize those changes within their
+respective roles. This can cause cause delays and times where features don't
+match up with what is installed or can be configured.
+
+There is also a concern that copying files from one location to another can
+cause cases where files are out of sync or versions don't align with what is
+installed and what is configured (is what we configured correct for the image
+version we installed).
+
+## Rationale
+As was piloted with the Template Service Broker ("TSB" for short), a role was
+created to process and configure OCP templates that were synchronized (currently
+still a manual process) from the origin repo where the component resides.
+
+This allowed any changes for the installation to be maintained by the developers
+and keep them from needing to be fluent with Ansible. It also afforded a level
+of abstraction from needing to keep in sync with TSB related changes and then
+translate them to changes within an Ansible role.
+
+However, there is still the gap of ensuring a role configures changes appropriately
+for the version of the component it is installing and we are violating the "DRY"
+principle by needing to sync files between repos.
+
+Prior to 3.4, Aggregated Logging solved this using a deployer pod. Everything
+was versioned and contained within this artifact that the Integration
+Services team provided and maintained. The deployer pod lived in the same
+repo as the components that it configured. This is something that is addressed
+again now with the Service Broker, Ansible Playbook Bundles ("APBs" for short)
+are configured in such a way that a containerized install/uninstall is done for
+the component.
+
+## Design
+We should describe a way to create a containerized playbook and role to mirror
```

Comment:
This creates a new management problem - lots and lots of images that have to have errata tracked.  It also means you can't go and get the "official" version of something.  It does reduce syncing, and keeps teams aligned with their config.

Reply:
> It also means you can't go and get the "official" version of something.

I'm not following on this. What do you mean by "official" and how would using an APB prevent us from providing that?